File: MiniDOMFinanziaria/Finanziaria/Consulenti/Consulenti.cs
using System;
using DMD;
using minidom.internals;

namespace minidom
{
    namespace internals
    {
        [Serializable]
        public sealed class CConsulentiClass : CModulesClass<Finanziaria.CConsulentePratica>
        {
            internal CConsulentiClass() : base("modCQSPDConsulenti", typeof(Finanziaria.CConsulentiPraticaCursor), -1)
            {
            }

            public Finanziaria.CConsulentePratica GetItemByUser(Sistema.CUser user)
            {
                if (user is null)
                    throw new ArgumentNullException("user");
                return GetItemByUser(DBUtils.GetID(user));
            }

            public Finanziaria.CConsulentePratica GetItemByUser(int id)
            {
                if (id == 0)
                    return null;
                foreach (Finanziaria.CConsulentePratica item in LoadAll())
                {
                    if (item.IDUser == id)
                        return item;
                }

                return null;
            }

            public Finanziaria.CConsulentePratica GetItemByName(string nome)
            {
                nome = DMD.Strings.Trim(nome);
                if (string.IsNullOrEmpty(nome))
                    return null;
                foreach (Finanziaria.CConsulentePratica c in LoadAll())
                {
                    if (c.Stato == ObjectStatus.OBJECT_VALID && DMD.Strings.Compare(c.Nome, nome, true) == 0)
                        return c;
                }

                return null;
            }
        }
    }

    public partial class Finanziaria
    {
        private static CConsulentiClass m_Consulenti = null;

        public static CConsulentiClass Consulenti
        {
            get
            {
                if (m_Consulenti is null)
                    m_Consulenti = new CConsulentiClass();
                return m_Consulenti;
            }
        }
    }
}

File: MiniDOMFinanziaria/Finanziaria/Consulenti/Consulenti.vb
Imports minidom
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica
Imports minidom.Internals
Imports minidom.Finanziaria

Namespace Internals

    <Serializable>
    Public NotInheritable Class CConsulentiClass
        Inherits CModulesClass(Of CConsulentePratica)

        Friend Sub New()
            MyBase.New("modCQSPDConsulenti", GetType(CConsulentiPraticaCursor), -1)
        End Sub

        Public Function GetItemByUser(ByVal user As CUser) As CConsulentePratica
            If (user Is Nothing) Then Throw New ArgumentNullException("user")
            Return GetItemByUser(GetID(user))
        End Function

        Public Function GetItemByUser(ByVal id As Integer) As CConsulentePratica
            If id = 0 Then Return Nothing

            For Each item As CConsulentePratica In Me.LoadAll
                If (item.IDUser = id) Then Return item
            Next


            Return Nothing
        End Function

        Public Function GetItemByName(nome As String) As CConsulentePratica
            nome = Strings.Trim(nome)
            If (nome = "") Then Return Nothing
            For Each c As CConsulentePratica In Me.LoadAll
                If (c.Stato = ObjectStatus.OBJECT_VALID AndAlso Strings.Compare(c.Nome, nome, CompareMethod.Text) = 0) Then Return c
            Next
            Return Nothing
        End Function
    End Class

End Namespace



Partial Public Class Finanziaria



    Private Shared m_Consulenti As CConsulentiClass = Nothing

    Public Shared ReadOnly Property Consulenti As CConsulentiClass
        Get
            If (m_Consulenti Is Nothing) Then m_Consulenti = New CConsulentiClass
            Return m_Consulenti
        End Get
    End Property

End Class